### DISCUSSION TITLE
Any way to not expose infuraId?

### DISCUSSION THREAD

**Discussion by sicadev** [WalletConnect/walletconnect-monorepo#1361] [Developer Support]
Hello,

I implemented walletconnect in a front end vue app but i don't see any way to not expose my infuraid.

I do have a backend to which i could request an infura jwt token instead but does walletconnect support this instead of infuraid directly?

Regards

**Comment by hackyguru** [comment]
I think using `infuraId` as an environment variable should be sufficient? 👀 

With WalletConnect, you can use any provider like Infura of your choice. I believe you'll need to specify the `infuraId` even if you use Infura directly without WalletConnect. Pls correct me if I'm wrong. 🙏

**Reply by christopheSeeka** [reply]
But walletconnect is used frontend so even with the infuraid in env variable, it will end up exposed in compiled front end code. If you look at opensea for example you can also find their infuraid in their js code on the front end, same with uniswap etc, doesnt seem to be something we can avoid to expose even if i wish we could. Or maybe im missing something?

**Comment by chadyj** [comment]
One solution is to use ALLOWLISTS with Infura to prevent 3rd party abuse.

See https://docs.infura.io/infura/networks/ethereum/how-to/secure-a-project
and
https://docs.infura.io/infura/networks/ethereum/how-to/secure-a-project/use-an-allowlist

**Reply by christopheSeeka** [reply]
Im aware of these options on infura side but it doesnt prevent curl request but i guess its the best available option yes, thanks :)

**Reply by svonian** [reply]
the best option would be to have the jwt support to send freshly signed tokens but do not see a way for wallet connect supporting this
